fili-core/src/main/java/com/yahoo/bard/webservice/logging/TimedPhase.java

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.logging;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Represents a phase that is timed.
+ * TimedPhase is used to associate a Timer located in the registry with the exact duration of such a phase for a
+ * specific request.
+ */
+public class TimedPhase implements AutoCloseable {
+    private static final Logger LOG = LoggerFactory.getLogger(TimedPhase.class);
+
+    private final String name;
+    private long start;
+    private long duration;
```

Comment:
I think `start` and `duration` need to be made thread-safe (likely with AtomicLongs)... It wasn't an issue before, because `TimedPhase` was encapsulated within the `RequestLog`, which was a `ThreadLocal` and was never able to exist on more than 1 thread at a time, but `TimedPhase` is now it's own _thing_, and there's a possibility of sharing them across threads.

I tried playing around with using `AtomicLong` for these variables a little bit, and it's not hard to retain the existing semantics, though it did make me think about if we _wanted_ to maintain the existing semantics around multiple consecutive starts / stops, especially in the new world of being across threads. No good solutions to this came to mind immediately, so it may make sense to add a cautionary note to the class-level doc for `TimedPhase` indicating the _possible_ race condition of starting and stopping the same phase on multiple threads.

Anyway, here's the relevant results of my exploration, in case they help:

```java
private static final long NOT_RUNNING = 0;

private final String name;
private final AtomicLong start = new AtomicLong();
private final AtomicLong duration = new AtomicLong();

/**
 * Start the phase.
 *
 * @return This phase after being started
 */
public TimedPhase start() {
    if (!start.compareAndSet(NOT_RUNNING, System.nanoTime())) {
        LOG.warn("Tried to start timer that is already running: {}", name);
    }
    return this;
}

/**
 * Stop the phase.
 */
public void stop() {
    long started = start.getAndUpdate(ignoredCurrent -> NOT_RUNNING);
    if (started == 0) {
        LOG.warn("Tried to stop timer that has not been started: {}", name);
        return;
    }
    duration.addAndGet(System.nanoTime() - started);
}

public long getDuration() {
    return duration.get();
}

public boolean isStarted() {
    return start.get() != 0;
}
```